isis/src/control/objs/ControlNet/ControlNet.cpp

```diff
@@ -1100,19 +837,34 @@ namespace Isis {
    * @returns A list of all measures which are in a given cube
    */
   QList< ControlMeasure * > ControlNet::GetMeasuresInCube(QString serialNumber) {
-    ValidateSerialNumber(serialNumber);
-    return (*cubeGraphNodes)[serialNumber]->getMeasures();
+    if( ValidateSerialNumber(serialNumber) ) {
+      return m_controlGraph[m_vertexMap[serialNumber]].measures.values();
+    }
+    else {
+      IString msg = "Cube Serial Number [" + serialNumber + "] not found in "
+          "the network";
+      throw IException(IException::Programmer, msg, _FILEINFO_);
+    }
```

Comment:
This will probably generate compiler warnings about possibly not returning. Make the exception in an if and then have the regular functionality after the if block.

Reply:
Like in FindClosest

Reply:
:+1: